bindata/manifests/machine-config/files/switchdev-vf-link-name.sh.yaml

```diff
@@ -0,0 +1,59 @@
+mode: 0755
+overwrite: true
+path: "/etc/udev/switchdev-vf-link-name.sh"
+contents:
+  inline: |
+    #!/bin/bash
```

Comment:
the switchdev-vf-link-name.sh is doing lot of logic which can case races. We saw it in openstack. It is better to do it like it done in os-net-config [1]. The script is just running simple echo command. We also mark the VF rep to be unmanaged by the NM with 'ENV{NM_UNMANAGED}="1" 


[1] https://github.com/openstack/os-net-config/blob/3144a8fc4fae95e3cfdc680ccdcbd5144a9a17a8/os_net_config/sriov_config.py#L344-L373
[2] - https://github.com/openstack/os-net-config/blob/3144a8fc4fae95e3cfdc680ccdcbd5144a9a17a8/os_net_config/sriov_config.py#L52-L58

Reply:
> the switchdev-vf-link-name.sh is doing lot of logic which can case races. We saw it in openstack. It is better to do it like it done in os-net-config [1]. The script is just running simple echo command. We also mark the VF rep to be unmanaged by the NM with 'ENV{NM_UNMANAGED}="1"
> 
> [1] https://github.com/openstack/os-net-config/blob/3144a8fc4fae95e3cfdc680ccdcbd5144a9a17a8/os_net_config/sriov_config.py#L344-L373
> [2] - https://github.com/openstack/os-net-config/blob/3144a8fc4fae95e3cfdc680ccdcbd5144a9a17a8/os_net_config/sriov_config.py#L52-L58

But Moshe in os-net-config, we know the name of the sriov_pf  and the switch_id and we add the udev rule according to that, Do we have such info here ? @moshe010

Reply:
> > the switchdev-vf-link-name.sh is doing lot of logic which can case races. We saw it in openstack. It is better to do it like it done in os-net-config [1]. The script is just running simple echo command. We also mark the VF rep to be unmanaged by the NM with 'ENV{NM_UNMANAGED}="1"
> > [1] https://github.com/openstack/os-net-config/blob/3144a8fc4fae95e3cfdc680ccdcbd5144a9a17a8/os_net_config/sriov_config.py#L344-L373
> > [2] - https://github.com/openstack/os-net-config/blob/3144a8fc4fae95e3cfdc680ccdcbd5144a9a17a8/os_net_config/sriov_config.py#L52-L58
> 
> But Moshe in os-net-config, we know the name of the sriov_pf and the switch_id and we add the udev rule according to that, Do we have such info here ? @moshe010

I think it's better to add a udev rule per pf here [1], you can see this one [2]
[1] https://github.com/k8snetworkplumbingwg/sriov-network-operator/blob/master/bindata/manifests/machine-config/files/configure-switchdev.sh.yaml
[2] https://github.com/Mellanox/switchdev-utils/blob/master/switchdev_utils/switchdev.sh#L12-L74

Reply:
switch_id is only available when switchdev mode is configured. If we were to add switch_id in udev rules, we also need to add logic in config daemon to reload udev rules which I intended to avoid. because appliance of udev rules will be done automatically along with rebooting (which is required to configure switchdev device) and config daemon doesn't have to track whether node is rebooted or not (to apply the udev rules). The current change looks simple than updating config daemon code since we handle all the switchdev config via plugins.

Reply:
Removed the long bash script for renaming vf rep

Reply:
@moshe010 @wmousa  have you seen any issue in openstack that udev rules cannot be applied due to vf rep is not link down?

Reply:
we did encounter  it. Are you saying that if VF rep is up udev rule will failed?

Reply:
yes

Reply:
but why the VF reps are up? They are not moved to the container namespace right?

Reply:
When the VF reps first get created, won't they be linked up auto?

Reply:
no they should be down. lined up auto if PF to VF.

Reply:
can you please check again? maybe NM starting them up?

Reply:
If it's NM, then I probably need to set the UM udev rules before any switchdev device is configured, otherwise if NM udev rules is set after rebooting, then it takes another reboot to take effect.
I'll split the rules to separate functions and call them differently.

Reply:
Did it solve the issue?

Reply:
I'm still testing it, what I observed was:
On first reboot, VF rep was not renamed correctly, still has the name ethx
On subsequential reboot, VF rep was renamed correctly.

Reply:
ok, the issue that vf was not renamed on first reboot is because udev rules are added after reboot so they are not re-loaded automatically. Fixed it by explicitly reload the udev rules before triggering.